pages/admin.js

```diff
@@ -378,15 +375,166 @@ function DeliveryTimes(props) {
                 <option>PM</option>
               </select>
             </div>
-
             <button type='submit' className='btn btn-outline p-2'>Add!</button>
           </form>
         </div>
       </div>
     </div>
+
+    
   )
 }
 
+function Categories(props) {
+  const [formData, setFormData] = useState({});
+  const [submitStatus, setSubmitStatus] = useState("");
+
+
+  const fetcher = (url) => fetch(url).then((res) => res.json())
+  let { data, error } = useSWR('/api/categories/ListCategories', fetcher)
+  
+  
+  if (error) {
+    console.log("Error fetching categories:", error)
+    return (
+      <div className='m-8'>
+        <div className='font-semibold text-3xl mb-4'>Categories</div>
+        <div className='m-4'>Error! See console log for details.</div>
+      </div>
+    )
+  }
+
+  if (!data) {
+    return (
+      <div className='m-8'>
+        <div className='font-semibold text-3xl mb-4'>Categories</div>
+        <div className='m-4'>Loading...</div>
+      </div>
+    )
+  }
+
+  if (Object.keys(formData).length == 0 && Object.keys(data).length > 0) {
+    setFormData(data)
+  }
+
+  const ref = firebase.database().ref('/category')
+
+  if (Object.keys(formData).length > 0) {
+    ref.on('child_added', snapshot => {
+      let data = snapshot.val()
+      if (!formData[data.tag]) {
+        setFormData({
+          ...formData, [data.tag]: data
+        });
+      }
+    });
+  }
+
+  let deleteCategory = (tag) => {
+    console.log(formData)
+    console.log(formData["categories"])
+    console.log(formData["categories"][0]["id"])
+    // console.log(tag)
```

Comment:
Delete these lines, we generally don't want console log messages unless necessary